.github/workflows/main.yml

```diff
@@ -70,3 +70,27 @@ jobs:
     - name: Run linting
       run: |
         pre-commit run --all-files
+    - name: Build output files
+      run: |
+        cd supply-chain
+        npm run build
+    - name: Check links in built files
+      id: link_check
+      run: |
+        cd supply-chain
+        find public -name "*.js" -exec grep -Eo "(http|https)://[a-zA-Z0-9./?=_%:()-]+" {} \; | sort -u > linklist.txt
```

Comment:
When I was looking at this again I realized we may need to include '&' to handle query params e.g. match all of something like `https://sciencemap.eto.tech/?china_affiliation_share=50%2C100&NP=50%2C13737&showList=1&cols=cluster_id%2Ccset_extracted_phrase%2Cclass_arts%2Cgrowth_3yr_p_rank%2CNP%2Cchina_affiliation_share%2Cindustry_share`

I wonder if in our case we could just get away with something like (I haven't tested this in bash, just python) `r"[\"\('](http|https://[^\"']+)[\"'\)]"` since links will appear surrounded by those characters in most or all cases we care about - but if you think we're good to go with the current regex, ok to keep it!

Reply:
Good idea. I think a broader regex will be useful here. I've edited your second regex slightly and added it in.